test-data/unit/check-enum.test

```diff
@@ -2105,3 +2109,31 @@ class AllPartialList(Enum):
 
     def check(self) -> None:
         reveal_type(self.value)  # N: Revealed type is "builtins.list[Any]"
+
+[case testEnumPrivateAttributeNotMember]
+from enum import Enum
+
+class MyEnum(Enum):
+    A = 1
+    B = 2
+    __my_dict = {A: "ham", B: "spam"}
+
+x: MyEnum = MyEnum.__my_dict  # E: Incompatible types in assignment (expression has type "Dict[int, str]", variable has type "MyEnum")
```

Comment:
Hmm, I'm not sure this is correct? The name is mangled at runtime, so this attempted attribute access fails with an exception unless you use the mangled name:

```pycon
>>> from enum import Enum
>>> class Foo(Enum):
...     __x = 'y'
...     y = 'z'
... 
>>> Foo.__x
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: type object 'Foo' has no attribute '__x'
>>> Foo._Foo__x
'y'
```

This PR definitely improves the handling of mangled names in enum class namespaces, but really mypy should be warning that `MyEnum` has no attribute `__my_dict` here (it has a `_MyEnum__my_dict` attribute instead). I think that's at least worth a comment, if it's not an easy fix.

Reply:
> Hmm, I'm not sure this is correct?

It isn't :D, but mypy doesn't understand name mangling yet and it is out of scope here. You are right however, there should be a comment.

Reply:
oof, we don't understand name mangling in any context? 😬

> You are right however, there should be a comment.

Thanks!

Reply:
We don't: https://mypy-play.net/?mypy=latest&python=3.12&gist=13d2e1e6e60b8e3828129fc44b336381 but there is already an issue for that #8267

Reply:
Oh and a PR #16715